### PR TITLE
💀 Wait for cleanup to die

### PR DESCRIPTION
**Pull request by Hacksore**
- **💀 Wait for cleanup to die**
- **👨🔬 "use the platform"**

### CODE REVIEW

File: packages/proxy/src/program.ts [hunk 1]
```diff
@@ -18,12 +19,27 @@ export async function program(opts: ProgramOptions) {
...
+  const stopWatching = createFileWatcher(fullyQualifiedPath, run);
   // Wrap up getting config and starting
   async function run() {
     const config = await getProxyConfig(fullyQualifiedPath);
-    await start(config);
+    const server = await start(config);
+
+    process.on("SIGINT", async () => {
+      // unset the proxy
+      await unsetProxy();
```
Comment (nickhudkins) [inline packages/proxy/src/program.ts:31]: all of these will go uncaught, which is "fine" but we could probably offer some nice debugg-ability (future us can do this, or perhaps hashtag good first issue)

File: packages/proxy/src/system.ts [hunk 2]
```diff
@@ -108,3 +106,27 @@ export function getNetworkAliasForMac(networkName: string) {
 
   return cleanedName;
 }
+
+// TODO: make a better impl if you feel like it but for now this works
```
Comment (nickhudkins) [inline packages/proxy/src/system.ts]: What's wrong with this? WHAT is there "TODO" here?

File: packages/proxy/src/system.ts [hunk 3]
```diff
@@ -108,3 +106,27 @@ export function getNetworkAliasForMac(networkName: string) {
...
+
+      // TODO: Cross Platform Support, and error handling.
+      if (platform() === "darwin") {
+        // make sure the interface is right because reasons
+        const networkInterfaceName = getNetworkAliasForMac(interfacesName);
+
+        const command = `networksetup -setautoproxyurl "${networkInterfaceName}" "off"`;
+        execSync(command);
+
+        console.log(`Proxy removed from [${interfacesName}]`);
```
Comment (nickhudkins) [inline packages/proxy/src/system.ts:126]: Do we not want some dope emoji? Or COLOR output?! (It's fine)

File: packages/proxy/src/utils.ts [hunk 4]
```diff
@@ -26,20 +26,48 @@ async function getFileHash(filePath: string) {
...
-    const configWatcher = fs.watch(filePath);
-    let previousFileHash = getFileHash(filePath);
-
-    for await (const event of configWatcher) {
-      if (event.eventType !== "change") continue;
-      const newFileHash = getFileHash(filePath);
-      if (previousFileHash !== newFileHash) {
-        onChange();
-        previousFileHash = newFileHash;
+function withResolvers() {
```
Comment (nickhudkins) [inline packages/proxy/src/utils.ts]: Nit: `withResolvers` here is "stolen" from `Promise.prototype.withResolvers` but as a freely named function `withResolvers` is a weird silly name. `createDeferred` is probably more appropriate. I don't care enough to reject this but a lovely cleanup task

File: packages/proxy/src/utils.ts [hunk 5]
```diff
@@ -26,20 +26,48 @@ async function getFileHash(filePath: string) {
...
+        const newFileHash = getFileHash(filePath);
+        if (previousFileHash !== newFileHash) {
+          onChange();
+          previousFileHash = newFileHash;
+        }
+      }
+    } catch (err: unknown) {
+      if (err instanceof Error && err.name === "AbortError") {
+        console.log("Stopped file watcher");
+        resolve({});
```
Comment (nickhudkins) [inline packages/proxy/src/utils.ts]: why are we resolving with an empty object and not `void`?

File: packages/proxy/src/utils.ts [hunk 6]
```diff
@@ -26,20 +26,48 @@ async function getFileHash(filePath: string) {
...
+        if (event.eventType !== "change") continue;
+        const newFileHash = getFileHash(filePath);
+        if (previousFileHash !== newFileHash) {
+          onChange();
+          previousFileHash = newFileHash;
+        }
+      }
+    } catch (err: unknown) {
+      if (err instanceof Error && err.name === "AbortError") {
+        console.log("Stopped file watcher");
```
Comment (nickhudkins) [inline packages/proxy/src/utils.ts:61]: cooler logging? Should we have (UGH... a `logger` interface 🥴)

**Review by nickhudkins — APPROVED** [review]

File: packages/proxy/src/system.ts (hunk 2 above)
Comment (Hacksore) [reply packages/proxy/src/system.ts]: ¯\_(ツ)_/¯

File: packages/proxy/src/system.ts (hunk 3 above)
Comment (Hacksore) [reply packages/proxy/src/system.ts:126]: i do actually! good call

File: packages/proxy/src/utils.ts (hunk 5 above)
Comment (Hacksore) [reply packages/proxy/src/utils.ts]: i brokith the types and bent the knee

File: packages/proxy/src/utils.ts (hunk 6 above)
Comment (Hacksore) [reply packages/proxy/src/utils.ts:61]: yeah we just use like winston or something idk

File: packages/proxy/src/program.ts (hunk 1 above)
Comment (Hacksore) [reply packages/proxy/src/program.ts:31]: added todo

File: packages/proxy/src/system.ts (hunk 2 above)
Comment (Hacksore) [reply packages/proxy/src/system.ts]: removing the line

**Pull request merged**